src/rendergraph/common/rendergraph/attributeset.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "backend/attributeset.h"
+
+namespace rendergraph {
+class AttributeSet;
+}
+
+class rendergraph::AttributeSet : public rendergraph::backend::AttributeSet {
+  public:
+    AttributeSet(std::initializer_list<Attribute> list, const std::vector<QString>& names);
+    const std::vector<Attribute>& attributes() const;
+};
+
+namespace rendergraph {
+template<typename... T>
+AttributeSet makeAttributeSet(const std::vector<QString>& names) {
+    return AttributeSet({(Attribute::create<T>())...}, names);
+}
```

Comment:
consider taking a `std::span<const QString>`. This would avoid the temporary vector allocation when used with a literal (which is basically exclusively how this type has been used in the code so far.)